### ISSUE TITLE
Is setting up a GPU compute worker on AWS doable?

### ISSUE THREAD

**Issue by bomb2peng** [codalab/codalab-competitions#2908]
Hi! I am organizing a competition where I need to run GPU inference codes in the scoring step. 
I set the YAML file to use the "codalab/codalab-legacy:gpu" docker environment and found that GPU is not available.
I assume that we need to provide a private compute worker with GPU capability and attach a queue to it.
We are working on setting up an AWS worker. However I am not sure whether this is doable at all?
If it is doable. Which docker image should be used on the AWS worker? I found multiple options [here ](https://github.com/codalab/competitions-v1-compute-worker).  Should I use the competitions-v1-nvidia-worker [here](https://hub.docker.com/r/codalab/competitions-v1-nvidia-worker/tags?page=1&ordering=last_updated)?

Another question is what is the relationship between the docker env specified in YAML and the compute worker docker? Which docker environment should I modify to support the scoring process where GPU inference is required?

Thanks a lot!

**Comment by Didayolo** [comment]
Hi,

Indeed, Codalab's default compute workers doesn't have GPU.

> I assume that we need to provide a private compute worker with GPU capability and attach a queue to it.

Exactly.

> We are working on setting up an AWS worker. However I am not sure whether this is doable at all?

I'm pretty sure it's doable!

> Which docker image should be used [...] ?

**You need Docker images for two different purposes:**
- The environment of the compute worker itself. For this I think `codalab/competitions-v1-nvidia-worker:v1.5-compat` is the right image to use.
- The environment in which the submissions will be run. `codalab/codalab-legacy:gpu` may be a good choice but you can fork it or provide your own image.


This Wiki entries should be useful:
- [Using your own compute workers](https://github.com/codalab/codalab-competitions/wiki/User_Using-your-own-compute-workers) (not GPU, the image Docker used here is not the right one)
- [Creating GPU workers on Google Cloud](https://github.com/codalab/codalab-competitions/wiki/Creating-GPU-Workers-on-Google-Cloud) (Google Cloud specific, you'll need to setup de VM by yourself).

I hope this helps!

**Comment by bomb2peng** [comment]
Thanks a lot for your help. We have worked this out, and it is indeed doable. We used the `codalab/competitions-v1-nvidia-worker:v1.5-compat` image to setup the worker, and used the  `codalab/codalab-legacy:gpu` (with some added libraries) to run submissions. 
Some helpful instructions are from [https://github.com/codalab/competitions-v1-compute-worker](https://github.com/codalab/competitions-v1-compute-worker). Followers of this topic should notice the **_If you want to run with GPU:_** part in order to run submissions with GPU.
